front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -190,6 +201,19 @@ export function AgentMessage({
   const [references, setReferences] = useState<{
     [key: string]: RetrievalDocumentType;
   }>({});
+  const [activeReferences, setActiveReferences] = useState<
+    { index: number; document: RetrievalDocumentType }[]
+  >([]);
+
+  function updateActiveReferences(
+    document: RetrievalDocumentType,
+    index: number
+  ) {
+    const existingIndex = activeReferences.find((r) => r.index === index);
```

Comment:
This logic can probably be removed if we send all known references always?

Reply:
Not sure what you mean; the cite block will only send the refs it's citing (not all the cited refs), and sometimes there are re-citations in which case activeReferences should not be updated. Maybe I missed smth?